node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -44,24 +41,132 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   implicit def executionContext: ExecutionContext = system.dispatcher
 
-  val peers: Vector[Peer]
+  val peersToCheckStack: mutable.Stack[Peer] = mutable.Stack.empty[Peer]
+
+  private def getPeersFromDnsSeeds: Vector[Peer] = {
+    val dnsSeeds = nodeAppConfig.network.dnsSeeds
+    val addresses = dnsSeeds
+      .flatMap(seed => {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  private val _peerServices: mutable.Map[Peer, ServiceIdentifier] =
-    mutable.Map.empty
+  private def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDB]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val filteredAddresses = addresses.filter(
+        nodeAppConfig.torConf.enabled || _.networkId != AddrV2Message.TOR_V3_NETWORK_BYTE)
+      val inetSockets = filteredAddresses.map(a => {
+        NetworkUtil.parseInetSocketAddress(a.address, a.port)
+      })
+      val peers =
+        inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+      peers
+    }
+    peersF
+  }
 
-  def peerServices: Map[Peer, ServiceIdentifier] = _peerServices.toMap
+  def getPeersFromConf: Vector[Peer] = {
+    val addresses = nodeAppConfig.peers
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
 
-  def setPeerServices(
+  lazy val peerConnectionScheduler: Cancellable =
+    system.scheduler.scheduleWithFixedDelay(initialDelay =
+                                              Duration(10, TimeUnit.SECONDS),
+                                            delay =
+                                              Duration(5, TimeUnit.SECONDS)) {
+      new Runnable() {
+        override def run(): Unit = {
+          val peersInDbCountF = PeerDAO().count()
+          peersInDbCountF.map(cnt =>
+            if (cnt > 10) peerConnectionScheduler.cancel())
+          if (peersToCheckStack.size < 8)
+            peersToCheckStack.pushAll(getPeersFromDnsSeeds)
+          val peers = for { _ <- 0 to 7 } yield peersToCheckStack.pop()
+          peers.foreach(peer => {
+            addPeer(peer, keepConnection = false)
+            initializePeer(peer)
+          })
+        }
+      }
```

Comment:
Numbers here are what I thought best. Limit of 10 to stop finding peers it arbitrary and we might want to change that. I put up a limit because otherwise it would go on indefinitely.